front/components/assistant/conversation/AttachmentCitation.tsx

```diff
@@ -0,0 +1,211 @@
+  type: "file";
+  preview?: string;
+  isUploading?: boolean;
+  spaceName?: never;
+  path?: never;
+  spaceIcon?: never;
+}
+
+interface NodeAttachmentCitation extends BaseAttachmentCitation {
+  type: "node";
+  spaceName: string;
+  path?: string;
+  spaceIcon?: React.ComponentType;
+  preview?: never;
+  isUploading?: never;
+}
+
+export type AttachmentCitation =
+  | FileAttachmentCitation
+  | NodeAttachmentCitation;
+
+type AttachmentCitationProps = {
+  attachmentCitation: AttachmentCitation;
+  onRemove?: () => void;
+  className?: string;
+};
+
+export function AttachmentCitation({
+  attachmentCitation,
+  onRemove,
+  className = "w-40",
+}: AttachmentCitationProps) {
+  const tooltipContent =
+    attachmentCitation.type === "file" ? (
+      attachmentCitation.title
+    ) : (
+      <div className="flex flex-col gap-1">
+        <div className="font-bold">{attachmentCitation.title}</div>
+        <div className="flex gap-1 pt-1 text-sm">
+          <Icon visual={attachmentCitation.spaceIcon} />
+          <p>{attachmentCitation.spaceName}</p>
+        </div>
+        <div className="text-sm text-muted-foreground dark:text-muted-foreground-night">
+          {attachmentCitation.path}
+        </div>
+      </div>
+    );
+
+  return (
+    <Tooltip
+      trigger={
+        <Citation
+          className={className}
+          href={attachmentCitation.sourceUrl ?? undefined}
+          isLoading={
+            attachmentCitation.type === "file" && attachmentCitation.isUploading
+          }
+          action={
+            onRemove && (
+              <CitationClose
+                onClick={(e) => {
+                  e.stopPropagation();
+                  onRemove();
+                }}
+              />
+            )
+          }
+        >
+          {attachmentCitation.type === "file" && attachmentCitation.preview && (
+            <CitationImage imgSrc={attachmentCitation.preview} />
+          )}
+          <CitationIcons>{attachmentCitation.visual}</CitationIcons>
+          <CitationTitle className="truncate text-ellipsis">
+            {attachmentCitation.title}
+          </CitationTitle>
+          {attachmentCitation.type === "node" && (
+            <CitationDescription className="truncate text-ellipsis">
+              <div className="flex items-center gap-1">
+                <span>{attachmentCitation.spaceName}</span>
+              </div>
+            </CitationDescription>
+          )}
+        </Citation>
+      }
+      label={tooltipContent}
+    />
+  );
+}
+
+export function contentFragmentToAttachmentCitation(
+  contentFragment: ContentFragmentType
+): AttachmentCitation {
+  if (contentFragment.contentNodeData) {
+    const { provider, nodeType } = contentFragment.contentNodeData;
+    const logo = getConnectorProviderLogoWithFallback({ provider });
+
+    const visual =
+      provider && ["webcrawler", "folder"].includes(provider) ? (
+        <Icon visual={logo} size="sm" />
```

Comment:
extremely nitpick but for the future reference, by default icon size is `sm` so no need to add 😄

Reply:
Yeah, not a big fan of default values so I always specify it, but fair will remove 👍🏼